pydantic/_internal/_decorators.py

```diff
@@ -218,3 +226,107 @@ def in_ipython() -> bool:
+            value_skipped = True
+            continue
+        if parameter.kind == Parameter.VAR_KEYWORD:
+            accepts_variable_kwargs = True
+            break
+        elif parameter.kind == Parameter.VAR_POSITIONAL:
+            raise TypeError  # TODO: Error message
+        if parameter_name in {'values', 'config', 'context'}:
+            kwargs.add(parameter_name)
+        else:
+            raise TypeError  # TODO: Error message
+    if accepts_variable_kwargs:
+
+        def wrapper(
+            __input_value: Any, *, data: Any, config: CoreConfig | None, context: Any, **future_kwargs: Any
+        ) -> Any:
+            return validator(__input_value, values=data, config=config, context=context, **future_kwargs)
+
+        return wrapper
+    elif not kwargs:
+
+        def wrapper(
+            __input_value: Any, *, data: Any, config: CoreConfig | None, context: Any, **future_kwargs: Any
+        ) -> Any:
+            return validator(__input_value)
+
+        return wrapper
+    elif kwargs == {'values'}:
+
+        def wrapper(
+            __input_value: Any, *, data: Any, config: CoreConfig | None, context: Any, **future_kwargs: Any
+        ) -> Any:
+            return validator(__input_value, values=data)
+
+        return wrapper
+    elif kwargs == {'context'}:
+
+        def wrapper(
+            __input_value: Any, *, data: Any, config: CoreConfig | None, context: Any, **future_kwargs: Any
+        ) -> Any:
+            return validator(__input_value, context=context)
+
+    elif kwargs == {'config'}:
+
+        def wrapper(
+            __input_value: Any, *, data: Any, config: CoreConfig | None, context: Any, **future_kwargs: Any
+        ) -> Any:
+            return validator(__input_value, config=config)
+
+        return wrapper
+    elif kwargs == {'values', 'context'}:
+
+        def wrapper(
+            __input_value: Any, *, data: Any, config: CoreConfig | None, context: Any, **future_kwargs: Any
+        ) -> Any:
+            return validator(__input_value, values=data, context=context)
+
+        return wrapper
+    elif kwargs == {'values', 'config'}:
+
+        def wrapper(
+            __input_value: Any, *, data: Any, config: CoreConfig | None, context: Any, **future_kwargs: Any
+        ) -> Any:
+            return validator(__input_value, values=data, config=config)
+
+        return wrapper
+    elif kwargs == {'context', 'config'}:
+
+        def wrapper(
+            __input_value: Any, *, data: Any, config: CoreConfig | None, context: Any, **future_kwargs: Any
+        ) -> Any:
+            return validator(__input_value, context=context, config=config)
+
+        return wrapper
+
+    # kwargs == {'values', 'field', 'config'}
+    def wrapper(  # type: ignore[no-redef]
+        __input_value: Any, *, data: Any, config: CoreConfig | None, context: Any, **future_kwargs: Any
+    ) -> Any:
+        return validator(__input_value, values=data, context=context, config=config)
```

Comment:
We need to make this an `else` and then handle the deprecated cases (`field` primarily)`.

Reply:
Done